src/main/scala/cognite/spark/v1/AssetsRelation.scala

```diff
@@ -115,6 +116,43 @@ class AssetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
 
   override def uniqueId(a: AssetsReadSchema): Long = a.id
 
+  implicit val upsertToUpdateTransformer: Transformer[AssetsUpsertSchema, AssetUpdate] =
+    Transformer
+      .define[AssetsUpsertSchema, AssetUpdate]
+      .withFieldComputed(
+        _.labels,
+        u =>
+          u.labels match {
+            case None => None
+            case _ => Some(LabelsOnUpdate(add = Some(u.labels.get.map(CogniteExternalId))))
+        })
+      .buildTransformer
+
+  implicit val createToUpdateTransformer: Transformer[AssetCreate, AssetUpdate] =
+    Transformer
+      .define[AssetCreate, AssetUpdate]
+      .withFieldComputed(
+        _.labels,
+        u =>
+          u.labels match {
+            case None => None
+            case _ => Some(LabelsOnUpdate(add = u.labels))
+        })
+      .buildTransformer
+
+  implicit val upsertToCreateTransformer: Transformer[AssetsUpsertSchema, AssetCreate] =
+    Transformer
+      .define[AssetsUpsertSchema, AssetCreate]
+      .withFieldComputed(
+        _.labels,
+        u =>
+          u.labels match {
+            case None => None
+            case _ => Some(u.labels.get.map(CogniteExternalId))
+          })
+      .withFieldComputed(_.name, u => u.name.get)
+      .buildTransformer
+
   def toAssetReadSchema(a: Asset): AssetsReadSchema =
```

Comment:
I will revisit these transformer like functions to see if they can be Transformer.